File: StudyBuddy/CourseView.swift
//
//  CourseView.swift
//  StudyBuddy
//
//  Created by Chananthida Sopaphol on 23/3/2568 BE.
//

import SwiftUI

struct CourseView: View {
    @State private var searchText: String = ""
    @State private var selectedCourses: Set<String> = []
    
    let courses = [
        "Mathematics",
        "Nuclear Engineering",
        "Software Design",
        "Computer Programming I",
        "Physics II",
        "Service Design",
        "Mobile Dev",
        "Thai Language"
    ]

    var filteredCourses: [String] {
        if searchText.isEmpty {
            return courses
        } else {
            return courses.filter { $0.localizedCaseInsensitiveContains(searchText) }
        }
    }

    var body: some View {
        VStack(alignment: .leading, spacing: 5) {
            Text("    Seclect Your Courses")
                .font(.title2).bold()
                .foregroundColor(.black)
                .padding(.top, 35)
            // Search Bar
            SearchBarView(searchText: $searchText, placeholder: "Search Courses")
            
            // Course List
            ScrollView {
                VStack(spacing: 12) {
                    ForEach(filteredCourses, id: \.self) { course in
                        CourseRow(course: course, isSelected: selectedCourses.contains(course)) {
                            if selectedCourses.contains(course) {
                                selectedCourses.remove(course)
                            } else {
                                selectedCourses.insert(course)
                            }
                        }
                    }
                }
                .padding(.horizontal, 15)
            }

            // Continue Button
            Button(action: {
                // Handle Continue Action
            }) {
                Text("Continue")
                    .bold()
                    .frame(maxWidth: .infinity)
                    .frame(height: 55)
                    .background(Color.blue)
                    .foregroundColor(.white)
                    .cornerRadius(23)
                    .padding(.horizontal, 15)
            }
        }
    }
}

// Course Row Component
struct CourseRow: View {
    let course: String
    let isSelected: Bool
    var onTap: () -> Void

    var body: some View {
        HStack {
            Text(course)
                .foregroundColor(.black)
            Spacer()
            if isSelected {
                Image(systemName: "checkmark.circle.fill")
                    .foregroundColor(.blue)
            }
        }
        .padding()
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(isSelected ? Color.blue.opacity(0.2) : Color.white)
        .cornerRadius(12)
        .overlay(
                    RoundedRectangle(cornerRadius: 12)
                        .stroke(Color.gray.opacity(0.25), lineWidth: 1) // Thicker border
                )
        .onTapGesture {
            onTap()
        }
    }
}

#Preview {
    CourseView()
}

File: frontend/components/CourseView.js
import React, { useState } from 'react';
import { View, Text, TextInput, TouchableOpacity, ScrollView, StyleSheet, FlatList } from 'react-native';
import CourseTag from './CourseTag';

const COURSES = [
  "Mathematics",
  "Nuclear Engineering",
  "Software Design",
  "Computer Programming I",
  "Physics II",
  "Service Design",
  "Mobile Dev",
  "Thai Language"
];

export default function CourseView() {
  const [searchText, setSearchText] = useState('');
  const [selectedCourses, setSelectedCourses] = useState([]);

  const filteredCourses = searchText
    ? COURSES.filter(course =>
        course.toLowerCase().includes(searchText.toLowerCase())
      )
    : COURSES;

  const toggleCourse = (course) => {
    setSelectedCourses(prev =>
      prev.includes(course)
        ? prev.filter(c => c !== course)
        : [...prev, course]
    );
  };

  return (
    <View style={styles.container}>
      <Text style={styles.title}>Select Your Courses</Text>
      <TextInput
        style={styles.input}
        placeholder="Search Courses"
        value={searchText}
        onChangeText={setSearchText}
        placeholderTextColor="#888"
      />
      <ScrollView style={styles.listContainer}>
        {filteredCourses.map(course => (
          <TouchableOpacity
            key={course}
            style={[
              styles.courseRow,
              selectedCourses.includes(course) && styles.selectedRow
            ]}
            onPress={() => toggleCourse(course)}
            activeOpacity={0.7}
          >
            <Text style={styles.courseText}>{course}</Text>
            {selectedCourses.includes(course) && (
              <Text style={styles.checkmark}>✔️</Text>
            )}
          </TouchableOpacity>
        ))}
      </ScrollView>
      <TouchableOpacity
        style={styles.continueButton}
        onPress={() => {
          // Handle Continue Action
        }}
        activeOpacity={0.8}
      >
        <Text style={styles.continueText}>Continue</Text>
      </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#f5f6fa',
    paddingTop: 35,
  },
  title: {
    fontSize: 22,
    fontWeight: 'bold',
    color: '#000',
    marginLeft: 17,
    marginBottom: 10,
  },
  input: {
    height: 50,
    backgroundColor: '#f1f1f6',
    borderRadius: 23,
    marginHorizontal: 17,
    paddingHorizontal: 16,
    fontSize: 16,
    marginBottom: 10,
    color: '#000',
  },
  listContainer: {
    marginHorizontal: 15,
    marginBottom: 20,
    maxHeight: 350,
  },
  courseRow: {
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: '#fff',
    borderRadius: 12,
    borderWidth: 1,
    borderColor: 'rgba(128,128,128,0.25)',
    padding: 16,
    marginBottom: 12,
  },
  selectedRow: {
    backgroundColor: 'rgba(37,99,235,0.2)',
  },
  courseText: {
    color: '#000',
    flex: 1,
    fontSize: 16,
  },
  checkmark: {
    color: '#2563eb',
    fontSize: 22,
    marginLeft: 8,
  },
  continueButton: {
    backgroundColor: '#2563eb',
    borderRadius: 23,
    marginHorizontal: 15,
    height: 55,
    justifyContent: 'center',
    alignItems: 'center',
    marginBottom: 20,
  },
  continueText: {
    color: '#fff',
    fontWeight: 'bold',
    fontSize: 18,
  },
});